trieste/bayesian_optimizer.py

```diff
@@ -444,45 +444,51 @@ def optimize(
         models = cast(
             Dict[str, TrainableProbabilisticModelType], map_values(create_model, model_specs)
         )
+
         history: list[Record[StateType]] = []
 
         for step in range(num_steps):
             set_step_number(step)
             try:
+
                 if track_state:
                     models_copy = copy.deepcopy(models)
                     acquisition_state_copy = copy.deepcopy(acquisition_state)
                     history.append(Record(datasets, models_copy, acquisition_state_copy))
 
-                if step == 0 and fit_initial_model:
-                    for tag, model in models.items():
-                        dataset = datasets[tag]
-                        model.update(dataset)
-                        model.optimize(dataset)
+                with timeit() as total_wallclock_timer:
+                    if step == 0 and fit_initial_model:
+                        with timeit() as initial_model_fitting_timer:
+                            for tag, model in models.items():
+                                dataset = datasets[tag]
+                                model.update(dataset)
+                                model.optimize(dataset)
```

Comment:
That looks like way too many levels of nesting, but i don't have a good alternative. @uri-granta any ideas?

Reply:
I think it's fine (though I used to write LISP code so maybe I'm not scared of nesting as much as I should be)

Reply:
this method overall looks pretty scary to me. half-way through it's hard to figure out at which level you are